test/Request/SerializerTest.php

```diff
@@ -266,35 +272,28 @@ public function messagesWithInvalidHeaders()
      */
     public function testDeserializationRaisesExceptionForMalformedHeaders($message, $exceptionMessage)
     {
-        $this->setExpectedException('UnexpectedValueException', $exceptionMessage);
-        $request = Serializer::fromString($message);
+        $this->expectException(UnexpectedValueException::class, $exceptionMessage);
+
+        Serializer::fromString($message);
     }
 
     public function testFromStreamThrowsExceptionWhenStreamIsNotReadable()
     {
-        $this->setExpectedException('InvalidArgumentException');
-        $stream = $this
-            ->getMockBuilder('Psr\Http\Message\StreamInterface')
-            ->getMock();
+        $stream = $this->createMock(StreamInterface::class);
+        $stream->method('isReadable')->will($this->returnValue(false));
 
-        $stream->expects($this->once())->method('isReadable')
-            ->will($this->returnValue(false));
+        $this->expectException(InvalidArgumentException::class);
 
         Serializer::fromStream($stream);
     }
 
     public function testFromStreamThrowsExceptionWhenStreamIsNotSeekable()
     {
-        $this->setExpectedException('InvalidArgumentException');
-        $stream = $this
-            ->getMockBuilder('Psr\Http\Message\StreamInterface')
-            ->getMock();
-
-        $stream->expects($this->once())->method('isReadable')
-            ->will($this->returnValue(true));
+        $stream = $this->createMock(StreamInterface::class);
+        $stream->method('isReadable')->will($this->returnValue(true));
+        $stream->method('isSeekable')->will($this->returnValue(false));
 
-        $stream->expects($this->once())->method('isSeekable')
-            ->will($this->returnValue(false));
```

Comment:
I'm not sure I understand why these mocks were rewritten. The combination of `getMockBuilder()` + `getMock()` works in PHPUnit 5.6+ and 6.0+, as does the syntax of `expects()`. The changes you have introduced loosen the expectations by removing the `expects($this->once())` constraints...